aleo-programs/staking_lite/src/main.leo

```diff
@@ -82,59 +93,125 @@ program staking_lite.aleo {
   }
 
   finalize initial_deposit(microcredits: u64) {
+    // Ensure program is initialized
+    let initialized: u8 = is_initalized.get_or_use(0u8, 0u8);
+    assert_eq(initialized, 1u8);
+
     let balance: u64 = total_balance.get_or_use(0u8, 0u64);
     let shares: u64 = total_shares.get_or_use(0u8, 0u64);
+    let pending: u64 = pending_withdrawal.get_or_use(0u8, 0u64);
     assert_eq(balance, 0u64);
     assert_eq(shares, 0u64);
+    assert_eq(pending, 0u64);
 
     total_balance.set(0u8, microcredits);
     total_shares.set(0u8, microcredits);
     delegator_shares.set(ADMIN, microcredits);
   }
 
   transition set_commission_percent(commission_rate: u128) {
-    assert_eq(self.caller, ADMIN);
+    assert_eq(self.signer, ADMIN);
```

Comment:
I changed all of the `self.caller` to `self.signer`, this way if someone writes a program that manages this program, it should still work with the original admin.

Reply:
I think that makes a lot of sense. We should change it back to `self.caller`